pangeo_forge/recipe.py

```diff
@@ -392,15 +405,20 @@ class NetCDFtoZarrSequentialRecipe(NetCDFtoZarrRecipe):
     """There is only one sequence of input files. Each file can contain
     many variables.
 
+    :param processed_input_urls: The inputs already used to generate the existing dataset.
     :param input_urls: The inputs used to generate the dataset.
     """
 
+    processed_input_urls: Iterable[str] = field(repr=False, default_factory=list)
     input_urls: Iterable[str] = field(repr=False, default_factory=list)
 
     def __post_init__(self):
         super().__post_init__()
-        input_pattern = ExplicitURLSequence(self.input_urls)
-        self._inputs = {k: v for k, v in input_pattern}
+        self.processed_input_nb = len(self.processed_input_urls)
+        input_pattern = ExplicitURLSequence(self.processed_input_urls + self.input_urls)
+        self._inputs = {
+            k: {"url": v, "processed": k < self.processed_input_nb} for k, v in input_pattern
```

Comment:
The implementation of `ExplicitURLSequence` leaks here, as `k` would not necessarily be an incrementing index. We could instead have:
```python
"processed": v in self.processed_input_urls
```

Reply:
Actually this is not true, since `self.input_urls` are the new URLs (they don't contain the already processed URLs), which is actually different than for the `NetCDFtoZarrMultiVarSequentialRecipe`.

Reply:
We could change `self.input_urls` so that it also includes the already processed URLs, this would make `NetCDFtoZarrSequentialRecipe` similar to `NetCDFtoZarrMultiVarSequentialRecipe`.